cherab/core/model/lineshape.pyx

```diff
@@ -367,6 +425,388 @@ cdef class StarkBroadenedLine(LineShapeModel):
+        if value == 'pi':
+            self._polarisation = PI_POLARISATION
+        elif value == 'sigma':
+            self._polarisation = SIGMA_POLARISATION
+        elif value == 'no':
+            self._polarisation = NO_POLARISATION
+        else:
+            raise ValueError('Select between "pi", "sigma" or "no", {} is unsupported.'.format(value))
+
+
+cdef class ZeemanTriplet(ZeemanLineShapeModel):
+    r"""
+    Simple Doppler-Zeeman triplet (Paschen-Back effect).
+
+    :param Line line: The emission line object for this line shape.
+    :param float wavelength: The rest wavelength for this emission line.
+    :param Species target_species: The target plasma species that is emitting.
+    :param Plasma plasma: The emitting plasma object.
+    :param str polarisation: Leaves only :math:`\pi`-/:math:`\sigma`-polarised components:
+                             "pi" - leave central component,
+                             "sigma" - leave side components,
+                             "no" - all components (default).
+    """
+
+    def __init__(self, Line line, double wavelength, Species target_species, Plasma plasma, polarisation='no'):
+
+        super().__init__(line, wavelength, target_species, plasma, polarisation)
+
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    @cython.initializedcheck(False)
+    @cython.cdivision(True)
+    cpdef Spectrum add_line(self, double radiance, Point3D point, Vector3D direction, Spectrum spectrum):
+
+        cdef double ts, sigma, shifted_wavelength, photon_energy, b_magn, component_radiance, cos_sqr, sin_sqr
+        cdef Vector3D ion_velocity, b_field
+
+        ts = self.target_species.distribution.effective_temperature(point.x, point.y, point.z)
+        if ts <= 0.0:
+            return spectrum
+
+        ion_velocity = self.target_species.distribution.bulk_velocity(point.x, point.y, point.z)
+
+        # calculate emission line central wavelength, doppler shifted along observation direction
+        shifted_wavelength = doppler_shift(self.wavelength, direction, ion_velocity)
+
+        # calculate the line width
+        sigma = thermal_broadening(self.wavelength, ts, self.line.element.atomic_weight)
+
+        # obtain magnetic field
+        b_field = self.plasma.get_b_field().evaluate(point.x, point.y, point.z)
+        b_magn = b_field.get_length()
+
+        if b_magn == 0:
+            # no splitting if magnetic filed strength is zero
+            if self._polarisation == NO_POLARISATION:
+                return add_gaussian_line(radiance, shifted_wavelength, sigma, spectrum)
+
+            return add_gaussian_line(0.5 * radiance, shifted_wavelength, sigma, spectrum)
+
+        # coefficients for intensities parallel and perpendicular to magnetic field
+        cos_sqr = (b_field.dot(direction.normalise()) / b_magn)**2
+        sin_sqr = 1. - cos_sqr
+
+        # adding pi component of the Zeeman triplet
+        if self._polarisation != SIGMA_POLARISATION:
```

Comment:
why is the logic inverted?

```
if self._polarisation != SIGMA_POLARISATION:
```

vs 

```
if self._polarisation == PI_POLARISATION:
```

The later would be clearer.

Reply:
Is it due to there being a 3rd valid state? If so it would be sensible to point this out in a comment.

Reply:
Yes, this is due to a 3rd valid state, `NO_POLARISATION`, which is default. In the `NO_POLARISATION` case we see both pi and sigma components of the multiplet. I will add a comment.

Reply:
Improved the comments.